server/fishtest/helpers.py

```diff
@@ -19,3 +22,23 @@ def diff_url(run):
             run["args"]["resolved_base"][:10],
             run["args"]["resolved_new"][:10],
         )
+
+
+def ok_hash(tc_ratio, hash):
+    # for historical reasons hash doesn't scale linearly between tc ratios 1-6, so only check 5+.
+    if tc_ratio < 5:
+        return (
+            True  # MTC and STC are assumed fine, could be handled if we really want to
+        )
+    # tc_ratio 6 has has 64 MB, and linearly for all above
+    target_hash = 64 * tc_ratio / 6
+    # if hash is outside of 5% of the target, emit warning upon submission.
+    return 0.95 <= hash / target_hash <= 1.05
+
+
+def reasonable_run_hashes(run):
+    # if this func returns false, then emit warning to user to verify hashes
+    base_hash = get_hash(run["args"]["base_options"])
+    new_hash = get_hash(run["args"]["new_options"])
+    tc_ratio = get_tc_ratio(run["args"]["tc"], run["args"]["threads"])
```

Comment:
we should probably add a comment that this doesn't work on asymetric-time tests (which are exceedingly rare)

and ofc i have no idea how nodestime works either

Reply:
> we should probably add a comment that this doesn't work on asymetric-time tests (which are exceedingly rare)
> 
> and ofc i have no idea how nodestime works either

We can disable all these cases in the worst case scenario.
Which IMO should be the way to go since we don't know what to do about nodes_time atm, but time odds could be trivially disabled. (semi-drafting this)

Reply:
Disabled from the warning*